python/openchange/provision.py

```diff
@@ -672,20 +693,175 @@ def accountcontrol(names, lp, creds, username=None, value=0):
+
+        print "[+] Group %s extended and enabled" % groupname
+    else:
+        print "[!] Group '%s' not found" % groupname
+
+
+def delete_group(names, lp, creds, groupname):
+    """
+    Remove openchange provision attributes from group
+    """
+    db = get_local_samdb(names, lp, creds)
+    group_dn = get_group_dn(db, names.domaindn, names.firstorg, groupname)
+    if not group_dn:
+        raise Exception("Group not found " + group_dn)
+    ldif = """
+dn: %(group_dn)s
+changetype: modify
+delete: mailNickName
+delete: legacyExchangeDN
+delete: proxyAddresses
+delete: msExchRecipientTypeDetails
+delete: msExchRecipientDisplayType
+"""
+    ldif = ldif % {"group_dn": group_dn}
+    db.modify_ldif(ldif)
+
+
+def update_group(names, lp, creds, groupname):
+    """
+    Update the attributes for the group, should be called when a group is switched from security to distribution.
+    :param lp: Loadparm context
+    :param creds: Credentials context
+    :param names: provision names object.
+    :param username: Name of user to extend
+    """
+    db = get_local_samdb(names, lp, creds)
+    group_dn = get_group_dn(db, names.domaindn, names.firstorg, groupname)
+
+    res = db.search(base=group_dn, scope=SCOPE_BASE, attrs=['legacyExchangeDN'])
+    if len(res) == 0:
+        raise Exception("Group " + groupname + " is not provisioned for OpenChange")
+
+    (recipient_type_details, recipient_display_type) = _group_recipient_type_details(db, group_dn)
+    ldif = """
+dn: %(group_dn)s
+changetype: modify
+replace: msExchRecipientTypeDetails
+msExchRecipientTypeDetails: %(recipient_type_details)s
+replace: msExchRecipientDisplayType
+msExchRecipientDisplayType: %(recipient_display_type)s
+"""
+    ldif = ldif % {"group_dn": group_dn,
+                   "recipient_type_details": recipient_type_details,
+                   "recipient_display_type": recipient_display_type}
+    db.modify_ldif(ldif)
+
+
+def _group_recipient_type_details(db, group_dn):
+    """
+    Returned tuple (recipient_type_details, recipient_display_type)
+    Values from http://blogs.technet.com/b/benw/archive/2007/04/05/exchange-2007-and-recipient-type-details.aspx
+    Display type for Universal Security Group does not work in outlook2010 so I am using the
+    one for distribution group
+    """
+    res = db.search(base=group_dn, scope=SCOPE_BASE, attrs=['groupType'])
+    if len(res) == 1:
+        groupType = int(res[0]['groupType'][0])
+        universal = groupType & 0x8
+        security = groupType & 0x80000000
+    else:
+        raise Exception("Group not found " + group_dn)
+
+    if security and universal:
+        return (1024, 1)
+    elif security and not universal:
+        # no value for this combination, using the previous one
+        return (1024, 1)
+    elif not security and universal:
+        return (256, 1)
+    elif not security and not universal:
+        return (512, 1)
+
+
+def _smtp_user_and_domain_by_dn(dn, name, mail, names):
```

Comment:
As you are returning a tuple here, I'd documentation as you have correctly done in `_group_recipient_type_details` function. It helps to readability for me and for you in 2 months.